### PR TITLE
feat: creates backend uses to integrate into resources page

### PR DESCRIPTION
**Pull request by MATE023**
### Summary

Adds Resource objects, can create, delete and get.

### Reference
[Asana link](https://app.asana.com/0/1202843173947642/1204326554113119/f)

### CODE REVIEW

File: util/validators.js [hunk 1]
```diff
@@ -264,6 +264,53 @@ module.exports.validateCreateTaskInput = (
...
+
+  const titleValidator = /^[a-zA-Z0-9- ]{6,50}$/i;
+
+  const linkValidator = /^https?:\/\/([0-9\x41-\x5A\x61-\x7A\xC0-\xF6\xF8-\xFF.-]){1,63}(\.)[0-9\x41-\x5A\x61-\x7A\xC0-\xF6\xF8-\xFF0-9]{1,63}(((\/)[0-9\x41-\x5A\x61-\x7A\xC0-\xF6\xF8-\xFF-=?#&]{1,63}){1,63})?(\/)?((\.)[0-9\x41-\x5A\x61-\x7A\xC0-\xF6\xF8-\xFF.-=?#&]{1,63})?$/gm;
+
+  //Other link valiadators, will take out later
+  //Without other countries characters: /^https?:\/\/([a-zA-Z0-9.-]){1,63}(\.)[a-zA-Z0-9]{1,63}(((\/)[a-zA-Z0-9-]{1,63}){1,63})?(\/)?((\.)[a-zA-Z0-9.]{1,63})?$/gm;
+  //With other chars in domain name /^https?:\/\/([0-9\x41-\x5A\x61-\x7A\xC0-\xF6\xF8-\xFF.-]){1,63}(\.)[a-zA-Z0-9]{1,63}(((\/)[a-zA-Z0-9-=?#&]{1,63}){1,63})?(\/)?((\.)[a-zA-Z0-9.-=?#&]{1,63})?$/gm
+
+  if (title.trim() === "") {
```
Comment (w-omar) [inline util/validators.js:285]: try to use truthy/falsey values if possible. ie instead of
`if(title.trim() === "")`
try
`if(!title.trim())

**Comment by MATE023** [comment]
It should work fine, but feel free to test more

**Comment by MATE023** [comment]
Should work but needs testing with new link validator

**Comment by MATE023** [comment]
Taking this out of draft since I think it should cover everything needed on the front end

**Review by w-omar — APPROVED** [review]

**Pull request merged**